### DISCUSSION TITLE
Reading and writing to a ZoneTree.FullTextSearch, ZoneTree DB file over network stream

### DISCUSSION THREAD

**Discussion by aley1** [ZoneTree/ZoneTree.FullTextSearch#10] [Q&A] [answered]
Hi,
Thanks for the amazing work done on ZoneTree.

A question, is it possible to modify ZoneTree/ZoneTree.FullTextSearch to read and write bytes on demand over the network connection to/from another server rather than copying the entire db file over to the client server to load into memory. Essentially the server will be hosting the db file and the client which does not have any disk, will just create a permanent connection to read/write with all DB functionality available. Let's assume its a high speed network and latency will be low.

Thanks.

**Comment by koculu** [comment, marked as answer]
Hi @aley1,

Thanks for your question!

ZoneTree doesn’t use a single database file—it manages multiple files dynamically. Directly accessing these files over a network isn’t feasible due to concurrency and data integrity challenges.

A better approach would be to expose ZoneTree’s functionality via an API on the server. This way, clients can interact with the database remotely without worrying about low-level file access, while maintaining full DB functionality.
